dask/dataframe/groupby.py

```diff
@@ -1681,6 +1681,14 @@ def get_group(self, key):
 
     @_aggregate_docstring()
     def aggregate(self, arg, split_every=None, split_out=1, shuffle=None):
+        split_out = 1 if split_out is None else split_out
+        if split_out is None:
+            warnings.warn(
+                "split_out=None is deprecated, please use a positive integer, "
+                "or allow the default of 1",
+                category=DeprecationWarning,
```

Comment:
Maybe `FutureWarning` instead as this is intended for end-users? 

```suggestion
                category=FutureWarning,
```

Reply:
Also, could we add a small test to make sure this warning is being emitted as expected?